src/quo/components/wallet/confirmation_progress/view.cljs

```diff
@@ -0,0 +1,81 @@
+            [quo.theme :as quo.theme]
+            [react-native.core :as rn]))
+
+(def ^:private max-progress 100)
+(def ^:private min-progress 0)
+
+(defn- calculate-box-state
+  [state counter index]
+  (cond
+    (and (= state :sending) (>= counter index) (< index 3))                 :confirmed
+    (and (= state :confirmed) (>= counter index) (< index 5))               :confirmed
+    (and (= state :finalising) (>= counter index) (< index 5))              :confirmed
+    (and (= state :finalising) (>= counter index) (> index 4) (< index 20)) :finalized
+    (and (= state :finalized) (>= counter index) (< index 5))               :confirmed
+    (and (= state :finalized) (>= counter index) (> index 4))               :finalized
+    (and (= state :error) (>= counter index) (< index 2))                   :error
+    :else                                                                   :pending))
+
+(defn- calculate-box-state-network-left
+  [state]
+  (cond
+    (= state :error)                                                     :error
+    (or (= state :confirmed) (= state :finalising) (= state :finalized)) :confirmed
+    :else                                                                :pending))
+
+(defn- calculate-box-state-network-right
+  [state]
+  (cond
+    (= state :error)
+    :error
+    (or (= state :confirmed)
+        (= state :finalising)
+        (= state :finalized))
+    :finalized
+    :else
+    :pending))
+
+(defn- calculate-progressed-value
+  [state progress-value]
+  (case state
+    :finalising progress-value
+    :finalized  max-progress
+    min-progress))
+
+(defn- progress-boxes
+  [{:keys [state counter total-box customization-color]}]
+  [rn/view
+   {:accessibility-label :mainnet-progress-box
+    :style               (style/progress-box-container true)}
+   (let [numbers (range 1 total-box)]
+     (doall (for [n numbers]
+              [progress-box/view
+               {:state               (calculate-box-state state counter n)
+                :customization-color customization-color
+                :key                 n}])))])
+
+(defn- progress-boxes-arbitrum-optimism
+  [{:keys [state bottom-large? customization-color progress-value]}]
+  [rn/view
+   {:accessibility-label :progress-box
+    :style               (style/progress-box-container bottom-large?)}
+   [progress-box/view
+    {:state               (calculate-box-state-network-left state)
+     :customization-color customization-color}]
+   [progress-box/view
+    {:state               (calculate-box-state-network-right state)
+     :full-width?         true
+     :progressed-value    (calculate-progressed-value state progress-value)
+     :customization-color customization-color}]])
+
+(defn- view-internal
+  [{:keys [network] :as props}]
+  (case network
+    :mainnet                 [progress-boxes props]
+    (or :arbitrum :optimism) [progress-boxes-arbitrum-optimism props]
```

Comment:
I think you can drop the "or" here

Reply:
Also remember case's docstring, they must be test constants, also since `or`  and  `case` are macros, it's highly probable that the code written here sometimes is not behaving as expected.